cid.go

```diff
@@ -26,10 +26,13 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"io"
 	"strings"
 
 	mbase "github.com/multiformats/go-multibase"
 	mh "github.com/multiformats/go-multihash"
+
+	cbg "github.com/whyrusleeping/cbor-gen"
```

Comment:
:sob:

But go doesn't make this easy, does it.

Reply:
The alternative was to have the codegen stuff spit out the methods it needs into the generated file, which i'm not opposed to.

Reply:
Hm. Honestly, that may be a better approach. That way, we can optimize _everything_.

Reply:
I agree with @Stebalien but even more intensely: I don't think this is viable at all.  Having go-cid import cbor packages seems almost backwards.  It would put cbor things on the transitive import requirements for not-at-all-cbor things and that's not okay.  An import like this from one of our most imported libraries would be a huge lock-in on that package as well, and I don't think that's something we should go into lightly, and probably isn't even intentional.

This _has to_ go in the generated file.  Not here.